Cython/Compiler/Symtab.py

```diff
@@ -46,23 +46,22 @@ def punycodify_name(cname, mangle_with=None):
     # if passed the mangle_with should be a byte string
     # modified from  PEP489
     try:
-        cname = cname.encode('ascii')
+        cname.encode('ascii')
     except UnicodeEncodeError:
-        pyrex_prefix = Naming.pyrex_prefix.encode('ascii')
         cname = cname.encode('punycode').replace(b'-', b'_')
+        if not isinstance(cname,str):
+            cname = cname.decode('ascii') # for Python
```

Comment:
You probably meant the comment to read "for Python 3", but I would suggest always doing this, i.e. just decode the byte string right after punycode-encoding it, also in Py2 (because, why not).

Reply:
Yes - that makes sense. I've made the change.